docs/designs/transaction-design.md

```diff
@@ -0,0 +1,222 @@
+
+### Database Schema
+
+#### Database  
+
+**Tables:**  
+- **TransactionHistory**:  
+  - `TransactionID` (Primary Key) - Unique identifier for each transaction record (UUID or GUID).
+  - `UserID` (Foreign Key) - References the `Users` table to identify the admin user associated with the transaction.
+  - `TransactionGUID`- Unique identifier for each transaction session (UUID or GUID). Used to correlate multiple entries for the same transaction (e.g., for the same user interacting multiple times within one session).
+  - `ItemID` (Foreign Key): References the `Items` table to identify the item associated with the transaction.
+  - `TransactionType`: VARCHAR(50), Type of transaction (e.g., Add, Checkout).
+  - `TotalAmount`: DECIMAL(10, 2), Total amount of the transaction.
+  - `TransactionDate`: DATETIME, Date and time when the transaction occurred.
+
+**SQL Schema:**
+```sql
+CREATE TABLE TransactionHistory (
+    TransactionID UNIQUEIDENTIFIER PRIMARY KEY,       
+    UserID INT,                                        
+    TransactionGUID UNIQUEIDENTIFIER,                  
+    ItemID INT,                                         
+    TransactionType VARCHAR(50),                          
+    TotalAmount DECIMAL(10, 2),                         
+    TransactionDate DATETIME DEFAULT GETDATE(),         
+    
+    -- Define foreign key constraints
+    FOREIGN KEY (UserID) REFERENCES Users(UserID),      
+    FOREIGN KEY (ItemID) REFERENCES Items(ItemID)       
+);
+
+-- Optionally, we may want to create an index for faster searching by TransactionGUID, UserID, and TransactionDate
+CREATE INDEX IX_TransactionHistory_TransactionGUID ON TransactionHistory(TransactionGUID);
+CREATE INDEX IX_TransactionHistory_UserID ON TransactionHistory(UserID);
+CREATE INDEX IX_TransactionHistory_TransactionDate ON TransactionHistory(TransactionDate);
+```
+
+**Stored Procedures:**
+- **RecordTransaction**:  
+  - Inserts a new transaction record into the `TransactionHistory` table.
+  - Parameters: `UserID`, `TransactionGUID`, `ItemID`, `TransactionType`, `TotalAmount`.
+  
+```sql
+CREATE PROCEDURE LogTransaction
+    @UserID INT,                     
+    @TransactionGUID UNIQUEIDENTIFIER, 
+    @ItemID INT,                   
+    @TransactionType VARCHAR(50),     
+    @TotalAmount DECIMAL(10, 2)       
+AS
+BEGIN
+    -- Insert a new record into the TransactionHistory table
+    INSERT INTO TransactionHistory (
+        TransactionID,
+        UserID,
+        TransactionGUID,
+        ItemID,
+        TransactionType,
+        TotalAmount,
+        TransactionDate
+    )
+    VALUES (
+        NEWID(),                     -- Generate a new GUID for the TransactionID
+        @UserID,                     
+        @TransactionGUID,           
+        @ItemID,                   
+        @TransactionType,             
+        @TotalAmount,               
+        GETDATE()                 
+    );
+END;
+```
+
+- **GetTransactionHistory**:
```

Comment:
This is a great suggestion. I would like to hold off on implementing these until we have the user stories and the figma designs for the transaction history
- do we want them by user, building, date, category, etc.
- are we going to do client side filtering or server side.